WordPress/gutenberg
# Add native block inserter onboarding tooltip

* **Gutenberg Mobile:** https://github.com/wordpress-mobile/gutenberg-mobile/pull/3529
* **WordPress iOS:** https://github.com/wordpress-mobile/WordPress-iOS/pull/16537
* **WordPress Android:** https://github.com/wordpress-mobile/WordPress-Android/pull/14683

## Description
Fixes #30624. Adds a tooltip for the block inserter as a part of a larger editor onboarding experience. 

A native implementation was added for the existing `Tooltip` component used on the web. This was done in an effort to (1) align the platforms and (2) avoid further duplicating tooltip logic. E.g. a duplicative tooltip component exists for the native [Cover media focal point picker](https://github.com/WordPress/gutenberg/blob/b955783c7b76a4518dbc47f867a4ff3cdf03f1d6/packages/components/src/focal-point-picker/tooltip/index.native.js#L24), and hopefully can one day be replaced by the `Tooltip` implementation in this PR.

The `Button` component [already renders the existing Tooltip component](https://github.com/WordPress/gutenberg/blob/b955783c7b76a4518dbc47f867a4ff3cdf03f1d6/packages/components/src/button/index.native.js#L188-L194). Because of this, it was not necessary to wrap the block inserter dropdown menu button in a new `Tooltip` element. Instead, this PR leverages `Button`'s `title` and `showTooltip ` props to display new "Tap to add content" tooltip required for the onboarding feature. One benefit of this approach is that we do not need to worry about forwarding a React ref through to `Button`. 

## How has this been tested?
Automated tests were not added in this PR, as the feature is an experiment and may not remain. However, I do plan to create a follow up PR to add tests for the `Tooltip` component if this PR merges. The work in this PR was manually tested via the following flow: 

1. Install the build from sibling WPiOS and WPAndroid PRs. 
1. Create/edit a post to launch the block editor. 
1. ℹ️ **Expected:** A "Tap to add content" tooltip is displayed attached to the block inserter button. 
1. Type on the keyboard. 
1. ℹ️ **Expected:** The tooltip remains visible and positioned correctly. 
1. Interact with any other portion of the UI aside from the keyboard or rotate the device. 
1. ℹ️ **Expected:** The tooltip is dismissed. 
1. Close the block editor. 
1. Relaunch the block editor. 
1. ℹ️ **Expected:** The tooltip is not displayed for subsequent editor launches. 

For Android, one can [clear out the WordPress app data](https://stackoverflow.com/a/8916939/378228) to re-test the flow. For iOS, the only option I am aware of is uninstalling and re-installing the app again. One thing to keep in mind is that the WPCom API places a limit on the number of times you may log in with a single account over a period of time. It will start a timeout if you log in too frequently. I'm uncertain of the number of attempts that triggers the timeout or how long the timeout lasts. 

## Screenshots 
| iOS | Android |
| --- | --- |
| ![ios-block-inserter-tooltip](https://user-images.githubusercontent.com/438664/118867255-7d3b8000-b8a8-11eb-9164-b0326f648dff.jpeg) | ![android-block-inserter-tooltip](https://user-images.githubusercontent.com/438664/118866638-cf2fd600-b8a7-11eb-9689-3fd39439f294.jpg) |

## Types of changes
New feature

## Checklist:
